MoppApp/MoppApp/ContainerViewController.swift

```diff
@@ -20,14 +20,47 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  *
  */
-class ContainerViewController : MoppViewController, ContainerActions {
-    var container: MoppLibContainer!
+protocol ContainerViewControllerDelegate: class {
+    func getDataFileCount() -> Int
+    func getContainerPath() -> String
+    func openContainer(afterSignatureCreated: Bool)
+    func getContainerFilename() -> String
+    func getDataFileFilename(index: Int) -> String
+    func getDataFileOriginFilename(index: Int) -> String
+    func isContainerEmpty() -> Bool
+    func removeDataFile(index: Int)
+}
+
+protocol SigningContainerViewControllerDelegate: class {
+    func startSigning()
+    func getSignaturesCount() -> Int
+    func getSignature(index: Int) -> Any
+    func removeSignature(index: Int)
+    func isContainerSignable() -> Bool
+}
+
+protocol CryptoContainerViewControllerDelegate: class {
+    func addAddressees()
+    func getAddressee(index: Int) -> Any
+    func getAddresseeCount() -> Int
+    func removeSelectedAddressee(index: Int)
+    func startEncrypting()
+}
+
+class ContainerViewController : MoppViewController, ContainerActions, PreviewActions {
+    
+    var containerViewDelegate: ContainerViewControllerDelegate! = nil
```

Comment:
No need to assign `nil` here explicitly as for the Implicitly Unwrapped Optionals, `nil` is the default value when initialized.